ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/WebClientOptionsFactory.java

```diff
@@ -32,12 +33,25 @@ public WebClientOptions createWebClientOptions(final Config config) {
         new WebClientOptions()
             .setDefaultPort(config.getDownstreamHttpPort())
             .setDefaultHost(config.getDownstreamHttpHost())
-            .setTryUseCompression(true);
+            .setTryUseCompression(true)
+            .setProxyOptions(proxyOptions().orElse(null));
 
     applyTlsOptions(clientOptions, config);
     return clientOptions;
   }
 
+  private static Optional<ProxyOptions> proxyOptions() {
+    var proxyHost = System.getProperty("http.proxyHost", "none");
+    var proxyPort = Integer.valueOf(System.getProperty("http.proxyPort", "80"));
```

Comment:
`Integer.valueOf` can throw NumberFormatException. catch block can be used to assign default value of 80 to proxy port.

Reply:
Ideally we should be able to cover all the options in `ProxyOptions` i.e. https://vertx.io/docs/apidocs/io/vertx/core/net/ProxyOptions.html . Specifying only host/port will work for unauthenticated proxy.

Reply:
Hello @usmansaleem, I added two commits to the PR